exercise/staff_views.py

```diff
@@ -324,27 +324,42 @@ class NextUnassessedSubmitterView(ExerciseBaseView, BaseRedirectView):
     access_mode = ACCESS.ASSISTANT
 
     def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
-        # Query submitters who have not been assessed yet.
         submitter = None
-        submitters = (UserProfile.objects
-            .filter(submissions__exercise=self.exercise)
-            .annotate(
-                count_assessed=Count(
-                    'submissions__id',
-                    filter=(Q(submissions__grader__isnull=False)),
-                ),
-            )
-            .filter(count_assessed=0)
-            .order_by('id'))
+
+        # submissions for this exercise
+        submissions = Submission.objects.filter(exercise=self.exercise)
+
+        # filter only the latest submissions
+        submissions = (submissions.annotate(latest=Max('submission_time'))
+           .filter(submission_time=F('latest'))
+           .order_by('latest'))
 
         previous_user_id = request.GET.get('prev')
```

Comment:
Just a thought, since the old code is changed to order by submission times, would it be more useful to set the previous timestamp to the `prev` GET query parameter instead of the user id?